src/compile/common.ts

```diff
@@ -114,7 +117,10 @@ export function formatMixins(model: Model, channel: Channel, format: string) {
         def.format = model.config().numberFormat;
         break;
       case TEMPORAL:
-        def.format = timeFormat(model, channel) || model.config().timeFormat;
+        if ((!fieldDef.timeUnit) || (fieldDef.timeUnit &&
+              fieldDef.timeUnit.toString().indexOf('quarter') === -1)) {
```

Comment:
we have a util function for this. Also, use the constant for quarter instead of the string.